CONTRIBUTING.md

```diff
@@ -31,7 +31,39 @@ To setup a local development environment we recommend reading [Building Grafana
 
 - Add tests relevant to the fixed bug or new feature.
 
-- Follow [PR and commit messages guidelines](#PR-and-commit-messages-guidelines)
+### High level checks
+
+- [ ] The pull request adds value and the impact of the change is in line with [Backend](https://github.com/grafana/grafana/tree/master/pkg) or [Frontend](https://github.com/grafana/grafana/tree/master/style_guides).
+- [ ] The pull request works the way it says it should do.
+- [ ] The pull request does not increase the Angular code base.
+  > We are in the process of migrating to React so any increment of Angular code is generally discouraged from. (there are a few exceptions)
+- [ ] The pull request closes one issue if possible and does not fix unrelated issues within the same pull request.
+- [ ] The pull request contains necessary tests.
+
+### Low level checks
+
+- [ ] The pull request contains a title that explains the PR. Follow [PR and commit messages guidelines](#Pull-Requests-titles-and-message)
+- [ ] The pull request contains necessary link(s) to issue(s). 
+- [ ] The pull request contains commits with commit messages that are small and understandable. Follow [PR and commit messages guidelines](#Pull-Requests-titles-and-message)
+- [ ] The pull request does not contain magic strings or numbers that could be replaced with an `Enum` or `const` instead.
+
+#### Bug specific checks
+
+- [ ] The pull request contains only one commit if possible.
+- [ ] The pull request contains `Closes: #Issue` or `Fixes: #Issue` in pull request description.
+
+### Frontend specific checks
+
+- [ ] The pull request does not contain uses of `any` or `{}` without comments describing why.
+- [ ] The pull request does not contain large React component that could easily be split into several smaller components.
+- [ ] The pull request does not contain back end calls directly from components, use actions and Redux instead.
```

Comment:
Is this really true? Recently in few PR we on purpose did the API interaction in a component (https://github.com/grafana/grafana/pull/18116, https://github.com/grafana/grafana/pull/17917) not to make simple . things to complex.

Reply:
IMO we shouldn't and I thought that was considered best practices. There's always exceptions though.

Reply:
@dprokop I do not have an issue with having calls directly in component if it does not need redux but we should at least have some wrapper around it (HOC, render prop or hook) to properly handle unmounting and cancelations. Especially in case of longer requests it can create problems like setting state on unmounted component or hitting browser request limits.